cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1954,6 +1958,45 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
             return withVoid(fut);
         }
 
+        /**
+         * Get the cluster Id of the Kafka cluster
+         * 
+         * @return
+         */
+        Future<ReconciliationState> kafkaGetClusterId() {
+            if (kafkaStatus.getClusterId() != null) {
+                // Cluster Id does not change
+                return Future.succeededFuture(this);
+            } else {
+                return adminClientSecrets()
+                    .compose(compositeFuture -> {
+                        log.debug("{}: Attempt to get clusterId", reconciliation);
+                        try {
+                            String bootstrapHostname = KafkaResources.bootstrapServiceName(this.name) + "." + this.namespace + ".svc:" + KafkaCluster.REPLICATION_PORT;
+                            log.debug("{}: Creating AdminClient for clusterId using {}", reconciliation, bootstrapHostname);
+                            Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostname, compositeFuture.resultAt(0), compositeFuture.resultAt(1), "cluster-operator");
+                            Promise<ReconciliationState> resultPromise = Promise.promise();
+                            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ReconciliationState>executeBlocking(
+                                future -> {
+                                    try {
+                                        kafkaStatus.setClusterId(kafkaAdmin.describeCluster().clusterId().get());
+                                        future.complete(this);
+                                    } catch (Throwable e) {
+                                        future.fail(e);
+                                    }
+                                },
+                                true,
+                                resultPromise);
+                            return resultPromise.future();
+                        } catch (KafkaException e) {
+                            log.warn("{}: Kafka exception getting clusterId {}", reconciliation, e.getMessage());
+                        }
+
+                        return Future.succeededFuture(this);
```

Comment:
I wonder if this does what you want it to do. I think this code would do the warning and return the succeeded future only in case the `adminClientProvider.createAdminClient(bootstrapHostname, compositeFuture.resultAt(0), compositeFuture.resultAt(1), "cluster-operator");` fails. But not when `kafkaAdmin.describeCluster().clusterId().get()` fails. When `kafkaAdmin.describeCluster().clusterId().get()` fails the whole reconciliation will fail.

If we really just want to pass the reocnciliation when this fails as discussed before, maybe you need to turn it around and have the whole try-catch block inside the `executeBlocking` section.

Reply:
`kafkaAdmin` doesn't seem to get closed.

Reply:
@scholzj ok, I'll move everything into the `executeBlocking`
@tombentley good catch!